versioning/versions/versions.go

```diff
@@ -31,7 +32,9 @@ type Version struct {
 }
 
 type Spec struct {
-	Versions []Version
+	Versions   []Version
+	Extensions []string
```

Comment:
instead of a list of strings, I rather have this explicitly as fields in the spec, so we can write something like this:
```
spec:
  versions:
    [...]
  extensions:
    from: {}  # from has no configurations
    retry:
      indent: '    '
```

Having config, which is detached from the extensions, is undesirable.

Reply:
Done